src/main/scala/cromwell/engine/workflow/WorkflowManagerActor.scala

```diff
@@ -92,4 +101,39 @@ class WorkflowManagerActor extends Actor with CromwellActor {
   private def idByWorkflow(workflow: WorkflowActorRef): Option[WorkflowId] = {
     workflowStore.toMap collectFirst {case (k, v) if v == workflow => k}
   }
+
+  private def restartIncompleteWorkflows(): Unit = {
+    // If the clob inputs for this workflow can be converted to JSON, return the JSON
+    // version of those inputs in a Some().  Otherwise return None.
+    def clobToJsonInputs(result: QueryWorkflowExecutionResult): Option[binding.WorkflowRawInputs] = {
+      result.jsonInputs.parseJson match {
+        case JsObject(rawInputs) => Option(rawInputs)
+        case x =>
+          log.error(s"Error restarting workflow ${result.workflowId}: expected JSON inputs, got '$x'.")
+          None
+      }
+    }
+    val restartableStates = Some(Seq(WorkflowSubmitted, WorkflowRunning))
+    // Attempt to restart all the workflows in restartable states whose clob raw inputs
+    // can successfully be converted to JSON.
+    case class RestartableWorkflow(id: WorkflowId, source: WdlSource, inputs: binding.WorkflowRawInputs)
+
+    val restartableWorkflows = for {
+      workflow <- dataAccess.query(states = restartableStates)
+      jsonInputs = clobToJsonInputs(workflow)
+      if jsonInputs.isDefined
+    } yield RestartableWorkflow(workflow.workflowId, workflow.wdlSource, jsonInputs.get)
+
+    val num = restartableWorkflows.length
+    val displayNum = if (num == 0) "no" else num
+    val plural = if (num == 1) "" else "s"
+    log.info(s"Found $displayNum workflow$plural to restart.")
+
+    if (num > 0) {
+      val ids = restartableWorkflows.map { _.id.toString }.sorted
+      log.info(s"Starting workflow ID$plural: " + ids.mkString(", "))
+    }
+
+    restartableWorkflows foreach { w => submitWorkflow(w.source, w.inputs, w.id) }
```

Comment:
This never attempts to create a `WorkflowActor` with an `InitialStore` as should be the case for restarts.